### PR TITLE
Implement GCI15 website setup assistant

### PR DESCRIPTION
**Pull request by YagoGG**
With this PR, a new tab is added in the UI, that guides the user to successfully configure [FOSSASIA's GCI15 website repo](https://github.com/fossasia/gci15.fossasia.org). This includes cloning the repo, and checking if it has correctly configured the remotes `upstream` and `origin`.
This should fullfill #2.
# Screenshot

Google Code-In Companion Screenshot
# Known bugs
- The button with ID `gci-website__button` doesn't update its `innerHTML` with the corresponding action properly
- Clicking the button to configure the remote `origin` won't work (stays on hold until #9 is merged, so the assistant can automatically set the remote as `https://github.com/<username>/gci15.fossasia.org`)

### CODE REVIEW

File: fossasia-gci-website.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+      // event)
+      updateStatuses();
+    }
+  });
+
+  // Display the dialog
+  ipcRenderer.send('openDialog', options);
+}
+
+pathField.oninput = function(e) {
```
Comment (roonyh) [inline fossasia-gci-website.js:140]: can we remove this? seems like you are not using it?

Comment (YagoGG) [inline fossasia-gci-website.js:140]: I left it there because you can still reach the text box's content by pressing <kbd>Tab</kbd> and modify it, so it still has some utility.  
However, it's not quite necessary, so we can remove it if you think it's not worth to have it there.

Comment (roonyh) [inline fossasia-gci-website.js:140]: ah. lets keep it.

File: fossasia-gci-website.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    // The repo isn't cloned yet
+    default:
+      gciWebButton.disabled = true;
+      gciWebButton.style.cursor = 'not-allowed';
+      gciWebButton.innerHTML = 'Loading...';
+
+      document.querySelector('#gci-website__loading').className += ' is-active'; // Display the loading spinner
+
+      Git.Clone.clone('https://github.com/fossasia/gci15.fossasia.org.git', path, null)
+      .then(function(repo, err) {
```
Comment (roonyh) [inline fossasia-gci-website.js:195]: Javascript promises are a little different to callbacks. They are supposed to replace callbacks.

So the nodegit callback here does not work the same way with promises.

A promise compatible callback takes two parameters. The **success** object and the **err**. In this case **success** object is `repo`.

This is how you should use them in promises.

``` javascript
Git.Clone.clone('https://github.com/fossasia/gci15.fossasia.org.git', path, null)
.then(function(repo) {
  // if operation succeeds we come here
  // repo stuff here
})
.catch(function(err){
  // if it fails we come here
  // error stuff
})
```

`then` and `catch` are special methods of the promise object returned from `Git.Clone.clone`. They will be called according to the success of the operation.

Basically thats how `callback(result, err)` api would transfer to promises.

You can actually think of them as regular try catch block.
